src/components/Monitor.js

```diff
@@ -17,18 +30,19 @@ const iframeStyle = {
 const Monitor = () => {
 	const doomframeRef = useRef(null);
 	const [switchedOn, setSwitchedOn] = useLocalStorage("monitor-on", false);
-	const [zIndices, setZIndices] = useState({});
+	const [zIndices, setZIndices] = useState([]);
 	const [achievements] = useLocalStorage("parameters", {
 		levelTwo: false,
 		hundredKills: false,
 		gameCompleted: false,
 	});
 
-	const handleApplicationFocus = (title) => {
-		const currentIndices = Object.values(zIndices);
-		const maxZIndex = currentIndices.length ? Math.max(...currentIndices) : 0;
-		setZIndices({ ...zIndices, [title]: maxZIndex + 1 });
-	};
+	const isDoomRunning = zIndices.at(-1) === Apps.Doom;
+
+	const handleApplicationFocus = (title) => setZIndices(previousValue => [
```

Comment:
I made indices much simpler.
Now it is just array. When you open or focus any app, then it is moved to the end of list.
CSS zIndex value is just the index of the app in the array, therefore it is always topmost visible.